pandas/core/arrays/datetimes.py

```diff
@@ -1752,7 +1840,7 @@ def maybe_convert_dtype(data, copy):
         # GH#18664 preserve tz in going DTI->Categorical->DTI
         # TODO: cases where we need to do another pass through this func,
         #  e.g. the categories are timedelta64s
-        data = data.categories.take(data.codes, fill_value=NaT)
+        data = data.categories.take(data.codes, fill_value=NaT)._values
```

Comment:
what's the reasoning here?  Another level of unpacking I guess.  Is there a non-private way to do this?

Reply:
This eliminates Indexes from the return type. It keeps everything as an array.